gap/TGQuotientSequences.gi

```diff
@@ -426,4 +426,655 @@ function(sequence)
+end );
+
+InstallMethod( IsSparse, [ IsTGQuotientSequencesAdjacencyMatrixObj and IsTGQuotientSequencesAdjacencyMatrixComponentRep ], 
+function(tgQSAdjMat)
+	return tgQSAdjMat!.sparse;
+end );
+
+InstallMethod( GetAdjacencyMatrix, [ IsTGQuotientSequencesAdjacencyMatrixObj and IsTGQuotientSequencesAdjacencyMatrixComponentRep ], 
+function(tgQSAdjMat)
+	return tgQSAdjMat!.adjMatrix;
+end );
+
+
+
+InstallMethod( PrintString, [ IsTGQuotientSequencesAdjacencyMatrixObj and IsTGQuotientSequencesAdjacencyMatrixComponentRep ], 
+function(tgQSAdjMat)
+	return Concatenation( "TGQuotientSequencesAdjacencyMatrix( ", 
+		PrintString(Signature(tgQSAdjMat)), ", ",
+		"boundByGenus = ", PrintString(GetBoundByGenus(tgQSAdjMat)), ", ",
+		"ListTGQuotients = ", PrintString(GetListTGQuotients(tgQSAdjMat)), ", ",
+		"MirrorSymmetries = ", PrintString(GetMirrorSymmetries(tgQSAdjMat)), ", ",
+		"sparse = ", PrintString(IsSparse(tgQSAdjMat)), ", ",
+		"adjMatrix = ", PrintString(GetAdjacencyMatrix(tgQSAdjMat)), 
+	")" );
+end );
+
+InstallMethod( PrintObj, [ IsTGQuotientSequencesAdjacencyMatrixObj and IsTGQuotientSequencesAdjacencyMatrixComponentRep ], 
+function(tgQSAdjMat)
+	Print(PrintString(tgQSAdjMat));
+end );
+
+
+# Saves translation groups in cache
+cachedGroup@ := MemoizeFunction(function(conderIndex)
+	local tgANDQuot;
+	tgANDQuot := ValueOption("tgANDQuot");
+	return AsTGSubgroup(TGTranslationGroup(tgANDQuot[1], tgANDQuot[2]));
+end);
+
+# Takes a list of integers and returns a new integer of concatenated integers in the list.
+toIndex@ := function(IndexLst)
+	return Int(JoinStringsWithSeparator(List(IndexLst, x -> String(x)),""));
+end;
+
+InstallGlobalFunction( TGQuotientSequencesAdjacencyMatrix,
+function(tg)
+	local signature, conderIndexLst, boundByGenus, lenConder, adjMat, i, j, k, l, 
+         Quot1, Quot2, genus1, genus2, switch1, switch2, GAMMA1, GAMMA2, sparse, F, 
+         D, DELTA, a, b, c, embDDELTA, rels, relsfull, G, Gplus, mspLst, tempRecNames;
+
+	# Raising errors in input:
+	# ------------------------
+
+	if not IsProperTriangleGroupObj(tg) then
+		Error("The first argument must be a ProperTriangleGroup object.");
+		return fail;
+	fi;
+
+	# ---------------
+    	# Needed objects:
+    	# ---------------
+
+	signature := Signature(tg);
+	conderIndexLst := ListTGQuotients(signature);
+	
+	# ---------
+    	# Options :
+    	# ---------
+
+	# option bound
+	boundByGenus := ValueOption("boundByGenus");
+	if boundByGenus = fail then
+		boundByGenus := 102; # Max. in Conder's list 101.
+	elif not IsPosInt(boundByGenus) then
+		Error(StringFormatted("BoundByGenus {} is not a positive integer.", boundByGenus));
+	        return fail;
+	fi;
+
+	# option sparsity
+    	sparse := ValueOption("sparse");
+    	if sparse = fail then
+    		sparse := false;
+    	elif not IsBool(sparse) then
+		Error(StringFormatted("The option sparse {} is not valid. It must be a boolean.", sparse));
+		return fail;
+    	fi;
+	
+	# -----------
+	# Initialize:
```

Comment:
Please add some more comments in this part and in some of the parts below (especially Tomas' code, if possible) to describe what you are doing.